src/main/scala/mesosphere/marathon/FeaturesConf.scala

```diff
@@ -1,18 +1,71 @@
 package mesosphere.marathon
 
-import org.rogach.scallop.{ScallopConf, ScallopOption}
+import org.rogach.scallop.ScallopConf
 
 trait FeaturesConf extends ScallopConf {
-
-  /**
-    * Indicates the http backend to use
-    */
-  lazy val featureAkkaHttpServiceBackend: ScallopOption[Boolean] = opt[Boolean](
-    "feature_akka_http_service_backend",
-    descr = "Enabled the new Akka HTTP Service Backend",
-    default = Some(false),
+  lazy val features = opt[String](
+    "enable_features",
+    descr = s"A comma-separated list of features. Available features are: ${Features.description}",
     required = false,
+    default = None,
     noshort = true,
-    hidden = true
+    validate = validateFeatures
   )
+
+  lazy val deprecatedFeatureKeys = opt[String](
+    "deprecated_features",
+    descr = "A comma-separated list of deprecated features to continue to enable",
+    required = false,
+    default = None,
+    noshort = true,
+    validate = validateDeprecatedFeatures)
+
+  lazy val availableFeatures: Set[String] = features.get.map(parseFeatures).getOrElse(Set.empty)
+
+  lazy val deprecatedFeatures: Set[DeprecatedFeatures.DeprecatedFeature] = deprecatedFeatureKeys.get match {
+    case Some(dfKeys) =>
+      val f = parseFeatures(dfKeys).map { dfKey =>
+        DeprecatedFeatures.all.find(_.key == dfKey).getOrElse { throw new RuntimeException("We should not see this") }
```

Comment:
How about
```scala
// DeprecatedFeatures.scala
object DeprecatedFeature {
  def unapply(rawFeature: String): Option[DeprecatedFeature] = all.find(_.key == rawFeature)
}

// FeaturesConf.scala
val features = parseFeatures(dfKeys).collect { case DeprecatedFeature(f) => f }
DeprecatedFeatures.warnOrFail(features)
features
```

Reply:
Also, I explicitly didn't want to silently drop keys that didn't match; if we see that here, then it's a bug in the code.

Reply:
I understand that. However, the validation already ensures this, right? Overall it is a little odd to so `parse` and then we have to do one more parsing step.

Reply:
Well, I'd call `.get` but if fails because of scapegoat.

I agree it's weird to do a `parse` and then another parsing step. You know what? Maybe we can fix this whole thing by just using a custom opt parser, so the assertion of the existence of some key is done during the argument parse phase, and we don't have to deal with string keys anymore.